sound/soc/sof/topology.c

```diff
@@ -1303,6 +1316,25 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
 		return ret;
 	}
 
+	if (swidget->num_pin_bindings) {
+		pin_binding = kcalloc(swidget->num_pin_bindings, sizeof(*pin_binding),
+				      GFP_KERNEL);
+		if (!pin_binding)
+			return -ENOMEM;
+
+		ret = sof_parse_token_sets(scomp, pin_binding, comp_pin_binding_tokens,
+					   ARRAY_SIZE(comp_pin_binding_tokens), priv->array,
+					   le32_to_cpu(priv->size), swidget->num_pin_bindings,
+					   sizeof(*pin_binding));
+		if (ret < 0) {
+			dev_err(scomp->dev, "failed to parse pin binding word tokens for %s\n",
+				w->name);
+				return ret;
+		}
+
+		swidget->pin_binding = pin_binding;
```

Comment:
do we need to free it in  sof_widget_unload ?

Reply:
updated.